### ISSUE TITLE
[feature] Make CSI-S3 compatible with ReadWriteMany?

### ISSUE THREAD

**Issue by LennardWesterveld** [CTrox/csi-s3#8]
Hi There,

Is it possible to make CSI-S3 compatible with ReadWriteMany storage type?

Kind regards,
Lennard

**Comment by segator** [comment]
I supose depending on mounter type
s3fs and goofys probably can work.

s3ql and s3backer will never support it

**Comment by floretan** [comment]
According to [this commit](https://github.com/ctrox/csi-s3/commit/95d7e48328a321b8a02b95f56c4b932b235e94b3), currently ReadWriteOnce is the only supported access mode.

**Comment by LennardWesterveld** [comment]
@floretan Thats correct i did try i do, and what @segator says it was working before this commit. Because i created the upgrade to csi 1.0, but it is somehow slow on syncing.

**Comment by floretan** [comment]
@LennardWesterveld thank you for the clarification. Is there something about CSI that prevents support for ReadWriteMany from being possible, or would it be possible to add support in the future? This would be a killer feature, as currently all options for ReadWriteMany volumes are either unstable (NFS) or don't support dynamic provisioning.

**Comment by LennardWesterveld** [comment]
I think if we make some test scenarios and make a switch based on mount type and make the validation better on that commit that you mentioned it is possible with this CSI. I talked lot with cloud providers about this topic and they al saying that they not going to support ReadWriteMany them selfs at least this year. Other users that are needing ReadWriteMany are doing the storage now on there own like, NFS indeed or a Rook.io cluster. I needed also a ReadWriteMany solution but i tried this CSI with S3 on Digitalocean and the read/write where very slow and for a Drupal site that is writing all the image styles  and reading all images from that mounted s3 directory was to slow for me so i stopped using this CSI for now. Didn't try the speed on SWS or Google Cloud.

**Comment by ramilexe** [comment]
@LennardWesterveld how did you solve issue with CSI S3 and drupal? I have the same case.

**Comment by floretan** [comment]
My colleague Janis created a similar solution focusing only on rclone, and we're using it for ReadWriteMany volumes with Drupal. You can find it here: https://github.com/wunderio/csi-rclone

**Comment by blecx** [comment]
Is there any progress on ReadWriteMany for s3fs. So I tested it out and it still not works. 

Thank you very much
